modules/data_team_helper/php/endpoints/conflicts.class.inc

```diff
@@ -0,0 +1,287 @@
+                    'PSCID',
+                    'text',
+                    'PSCID',
+                    false,
+                    null
+                ),
+                'FieldName'         => parent::formElementObjectCreator(
+                    'FieldName',
+                    'text',
+                    'Field Name',
+                    false,
+                    null
+                ),
+                'TableName'         => parent::formElementObjectCreator(
+                    'TableName',
+                    'text',
+                    'TableName',
+                    true,
+                    null
+                ),
+                'test_name_display' => parent::formElementObjectCreator(
+                    'test_name_display',
+                    'text',
+                    'test_name_display',
+                    true,
+                    null
+                ),
+                'visit_label'       => parent::formElementObjectCreator(
+                    'visit_label',
+                    'select',
+                    'Visit',
+                    false,
+                    $visit_array
+                ),
+                'SessionID'   => parent::formElementObjectCreator(
+                    'SessionID',
+                    'text',
+                    'SessionID',
+                    true,
+                    null
+                ),
+                'Full_name'   => parent::formElementObjectCreator(
+                    'Full_name',
+                    'text',
+                    'Full_name',
+                    true,
+                    null
+                ),
+                'Instrument'        => parent::formElementObjectCreator(
+                    'Instrument',
+                    'select',
+                    'Instrument',
+                    false,
+                    \Utility::getAllInstruments()
+                ),
+                'Project'           => parent::formElementObjectCreator(
+                    'Project',
+                    'select',
+                    'Project',
+                    true,
+                    \Utility::getProjectList()
+                ),
+                'Site'              => parent::formElementObjectCreator(
+                    'Site',
+                    'select',
+                    'Site',
+                    true,
+                    $list_of_sites
+                ),
+               );
+    }
+
+    /**
+     * Permission verification.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
+            || !$user->hasPermission('genomic_browser_view_allsites')
```

Comment:
It seems like there's a lot of copy-pasting happening. You should consider whether there's a way to refactor all this boilerplate code into the parent class.

For example the parent could have a `_verifyPermissions` class that child classes can inherit. They don't need to declare their own that way unless they have something different to do.